modules/upi-provisioning-new-hosts-in-a-upi-cluster.adoc

```diff
@@ -22,42 +22,190 @@ To provision bare-metal hosts to the cluster by using the BMO, you must set the
+  preprovisioningNetworkDataName: openshift-worker-<num>-network-config-secret <1>
+----
++
+--
+<1> Replace `<num>` for the machine and worker numbers of the bare metal nodes in the `name`, `credentialsName`, and the `preprovisioningNetworkDataName` fields.
+<2> Add the NMState YAML syntax to configure the host interfaces. To configure the network interface for a newly created node, specify the name of the secret that contains the network configuration. Follow the `nmstate` syntax to define the network configuration for your node. See link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.15/html/installing/deploying-installer-provisioned-clusters-on-bare-metal#configuring-host-network-interfaces-in-the-install-config-yaml-file_ipi-install-installation-workflow[Optional: Configuring host network interfaces] for details on configuring NMState syntax.
+<3> Optional: If you have configured the network interface with `nmstate`, and you want to disable an interface, set `state: up` with the IP addresses set to `enabled: false` as shown:
++
+[source,yaml]
+----
+---
+   interfaces:
+   - name: <nic_name>
+     type: ethernet
+     state: up
+     ipv4:
+       enabled: false
+     ipv6:
+       enabled: false
+----
+<4> Replace `<nic1_name>`, `<ip_address>`, `<dns_ip_address>`, `<next_hop_ip_address>`, and `<next_hop_nic1_name>` with appropriate values.
+<5> Replace `<base64_of_uid>` and  `<base64_of_pwd>` with the base64 string of the user name and password.
+<6> Replace `<nic1_mac_address>` with the MAC address of the bare metal node's first NIC. See the "BMC addressing" section for additional BMC configuration options.
+<7> Replace `<protocol>` with the BMC protocol, such as IPMI, RedFish, or others. Replace `<bmc_url>` with the URL of the bare metal node's baseboard management controller.
+<8> To skip certificate validation, set `disableCertificateVerification` to true.
+<9> Replace `<bmc_username>` and `<bmc_password>` with the string of the BMC user name and password.
+<10> Optional: Replace `<root_device_hint>` with a device path if you specify a root device hint.
+--
 
+* *DHCP configuration* `bmh.yaml`:
++
+[source,yaml]
+----
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: openshift-worker-<num>-bmc-secret <1>
+  namespace: openshift-machine-api
+type: Opaque
+data:
+  username: <base64_of_uid> <2>
+  password: <base64_of_pwd> <2>
+---
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: openshift-worker-<num> <1>
+  namespace: openshift-machine-api
+spec:
+  online: True
+  bootMACAddress: <nic1_mac_address> <3>
+  bmc:
+    address: <protocol>://<bmc_url> <4>
+    credentialsName: openshift-worker-<num>-bmc-secret <1>
+    disableCertificateVerification: True <5>
+    username: <bmc_username> <6>
+    password: <bmc_password> <6>
+  rootDeviceHints:
+    deviceName: <root_device_hint> <7>
+  preprovisioningNetworkDataName: openshift-worker-<num>-network-config-secret
```

Comment:
This secret is not present in the DHCP example.

Reply:
Updated As Suggested.